### PR TITLE
Update to monaco-languageclient 6.5.0 and monaco-vscode-api 1.82.x

### PR DESCRIPTION
**Pull request by kaisalmen**
`monaco-vscode-api` now supplies all services as independent packages. Therefore the new `monaco-languageclient` version only allows to import services directly via `userConfig` and remove all the extra "enable" and "configure" params. By this approach all the extra dynamic imports are no longer present in application bundles. Only the things really used and specified as dependencies are used now.

It is now possible to specify multiple extensions and it is possible to await other extensions before loading your own.

**Important**: This still uses a next version of  `monaco-languageclient`. Will release and use a final versions once everything is tested.

### CODE REVIEW

**Comment by kaisalmen** [comment]
@cdietrich this should be interesting for you. Bundle sizes are now decreased.

File: package.json [hunk 1]
```diff
@@ -1,11 +1,11 @@
 {
-  "name": "monaco-components",
```
Comment (montymxb) [inline package.json:2]: Why was the package name dropped?

File: package.json [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
+  "type": "module",
   "private": true,
   "scripts": {
     "clean": "npm run clean --workspaces",
     "clean:watch": "tsc --build tsconfig.build.json --clean",
-    "dev": "vite",
-    "dev:debug": "vite --debug --force",
+    "dev": "node --experimental-import-meta-resolve ./node_modules/vite/bin/vite.js",
+    "dev:debug": "node --experimental-import-meta-resolve ./node_modules/vite/bin/vite.js --debug --force",
```
Comment (montymxb) [inline package.json:8]: I think even with experimental feature enabled it still gives a warning right?

File: packages/examples/src/langium/config/wrapperLangiumClassic.ts [hunk 3]
```diff
@@ -15,14 +20,11 @@ export const setupLangiumClientClassic = async (): Promise<UserConfig> => {
         },
         wrapperConfig: {
             serviceConfig: {
-                enableModelService: true,
-                configureEditorOrViewsService: {
+                userServices: {
+                    ...getConfigurationServiceOverride(Uri.file('/workspace')),
+                    ...getEditorServiceOverride(useOpenEditorStub),
+                    ...getKeybindingsServiceOverride()
```
Comment (montymxb) [inline packages/examples/src/langium/config/wrapperLangiumClassic.ts:26]: Just a note, should document this breaking change.

File: packages/examples/src/langium/config/wrapperStatemachineConfig.ts [hunk 4]
```diff
@@ -17,56 +25,52 @@ export const createLangiumGlobalConfig = async (htmlElement: HTMLElement): Promi
...
-                extensionFilesOrContents: extensionFilesOrContents,
+                    filesOrContents: extensionFilesOrContents
+                }],
                 userConfiguration: {
-                    json: `{
-    "workbench.colorTheme": "Default Dark Modern",
-    "editor.guides.bracketPairsHorizontal": "active",
-    "editor.lightbulb.enabled": true
-}`
+                    json: JSON.stringify({
```
Comment (montymxb) [inline packages/examples/src/langium/config/wrapperStatemachineConfig.ts:69]: This feels better than the raw JSON text from before.

File: packages/examples/src/reactTs.tsx [hunk 5]
```diff
@@ -18,7 +18,9 @@ const EditorDemo: React.FC = () => {
         htmlElement: rootElem,
         wrapperConfig: {
             serviceConfig: {
-                enableKeybindingsService: true,
+                userServices: {
+                    ...getKeybindingsServiceOverride()
```
Comment (montymxb) [inline packages/examples/src/reactTs.tsx:22]: Just a question, is there a risk of services overriding properties contributed from each other?

File: packages/examples/src/wrapperTs.ts [hunk 6]
```diff
@@ -1,18 +1,18 @@
...
-
-import { buildWorkerDefinition } from 'monaco-editor-workers';
+import { disposeEditor, startEditor, swapEditors, updateModel, wrapper } from './common.js';
 import { UserConfig } from 'monaco-editor-wrapper';
+import { buildWorkerDefinition } from 'monaco-editor-workers';
+
 buildWorkerDefinition('../../../node_modules/monaco-editor-workers/dist/workers', import.meta.url, false);
 
-const codeUri = '/tmp/hello.ts';
+const codeUri = '/workspace/hello.ts';
```
Comment (montymxb) [inline packages/examples/src/wrapperTs.ts:10]: Was there a reason for changing from tmp to workspace for all these cases?

File: packages/monaco-editor-wrapper/test/languageClientWrapper.test.ts [hunk 7]
```diff
@@ -96,7 +96,7 @@ describe('Test LanguageClientWrapper', () => {
         const languageClientConfig: LanguageClientConfig = {
             options: {
                 $type: 'WorkerConfig',
-                url: new URL('http://localhost:63315'),
+                url: new URL('http://localhost:5173'),
```
Comment (montymxb) [inline packages/monaco-editor-wrapper/test/languageClientWrapper.test.ts:99]: Arbitrary port change?

File: vite.config.ts [hunk 8]
```diff
@@ -1,53 +1,69 @@
...
+        },
+        optimizeDeps: {
+            esbuildOptions: {
+                plugins: [
+                    // copied from "https://github.com/CodinGame/monaco-vscode-api/blob/main/demo/vite.config.ts"
+                    {
+                        name: 'import.meta.url',
+                        setup({ onLoad }) {
+                            // Help vite that bundles/move files in dev mode without touching `import.meta.url` which breaks asset urls
+                            onLoad({ filter: /.*\.js/, namespace: 'file' }, async args => {
```
Comment (montymxb) [inline vite.config.ts:42]: This is the import.meta workaround right?

**Review by montymxb — COMMENTED** [review]
Some general notes, mostly questions. Ran locally with tests & examples, all looks good.

File: package.json (hunk 1 above)
Comment (kaisalmen) [reply package.json:2]: Not intended. Will bring it back. Interestingly a name is not required for the workspace package.

File: package.json (hunk 2 above)
Comment (kaisalmen) [reply package.json:8]: No really, this is just required for the copied vite plugin / the import.meta workaround.

File: packages/examples/src/langium/config/wrapperStatemachineConfig.ts (hunk 4 above)
Comment (kaisalmen) [reply packages/examples/src/langium/config/wrapperStatemachineConfig.ts:69]: I thought so, too.

File: packages/examples/src/langium/config/wrapperLangiumClassic.ts (hunk 3 above)
Comment (kaisalmen) [reply packages/examples/src/langium/config/wrapperLangiumClassic.ts:26]: Good point, I documented this in `monaco-languageclient` and just updated the CHANGELOG.

File: packages/examples/src/reactTs.tsx (hunk 5 above)
Comment (kaisalmen) [reply packages/examples/src/reactTs.tsx:22]: You can a service already specified before being initialized, but the parameters a unique to each single service.

File: packages/examples/src/wrapperTs.ts (hunk 6 above)
Comment (kaisalmen) [reply packages/examples/src/wrapperTs.ts:10]: `/tmp` felt temporary. 😆  `/workspace` is somewhat nicer, IMO.

File: packages/monaco-editor-wrapper/test/languageClientWrapper.test.ts (hunk 7 above)
Comment (kaisalmen) [reply packages/monaco-editor-wrapper/test/languageClientWrapper.test.ts:99]: You can specify a server port for vite, but not for vitest or it is not taken into account. This is somewhat stupid. An the latest minor version change updated the default port. Without this change the test fail in the pipeline. I am not happy with it, but currently there is no other way.

File: vite.config.ts (hunk 8 above)
Comment (kaisalmen) [reply vite.config.ts:42]: Exactly.

File: package.json (hunk 1 above)
Comment (kaisalmen) [reply package.json:2]: Done ⬇️

File: packages/examples/src/langium/config/wrapperLangiumClassic.ts (hunk 3 above)
Comment (kaisalmen) [reply packages/examples/src/langium/config/wrapperLangiumClassic.ts:26]: Done ⬇️

**Review by montymxb — APPROVED** [review]
:octocat: Alright approved!

File: packages/monaco-editor-wrapper/package.json [hunk 9]
```diff
@@ -58,10 +58,11 @@
     "npm": "9.8.1"
   },
   "dependencies": {
-    "monaco-languageclient": "~6.4.6"
+    "@codingame/monaco-vscode-configuration-service-override": "~1.82.2",
+    "monaco-languageclient": "6.5.0"
   },
   "peerDependencies": {
-    "monaco-languageclient": "~6.4.6",
+    "monaco-languageclient": "6.5.0s",
```
Comment (Yokozuna59) [inline packages/monaco-editor-wrapper/package.json:65]: Sorry to disturb, but is this version indented? It has `s` at the end, and there no such version for `monaco-languageclient`:

https://www.npmjs.com/package/monaco-languageclient?activeTab=versions

Comment (montymxb) [reply packages/monaco-editor-wrapper/package.json:65]: Nope definitely not, that's a mistake that I must've missed when reviewing. Thanks for pointing that out, we can get a patch release on that shortly.

Comment (kaisalmen) [reply packages/monaco-editor-wrapper/package.json:65]: @Yokozuna59 thanks for spotting this mistake. Will release a new version in a couple of minutes.

Comment (Yokozuna59) [reply packages/monaco-editor-wrapper/package.json:65]: @montymxb @kaisalmen No worries. Thanks for your fast response.

Comment (kaisalmen) [reply packages/monaco-editor-wrapper/package.json:65]: Done:
https://www.npmjs.com/package/monaco-editor-wrapper/v/3.2.2
https://www.npmjs.com/package/@typefox/monaco-editor-react/v/2.2.2

**Pull request merged**